test/unit-test/cmock_build.cmake

```diff
@@ -3,7 +3,7 @@ macro( clone_cmock )
         find_package( Git REQUIRED )
         message( "Cloning submodule CMock." )
         execute_process( COMMAND rm -rf ${CMOCK_DIR}
-                         COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive ${CMOCK_DIR}
+                         COMMAND ${GIT_EXECUTABLE} submodule update checkout --init --recursive ${CMOCK_DIR}
```

Comment:
Does this need to be --checkout?

Reply:
If we take out the clone you added to the ci.yml, we could test this?

Reply:
>Does this need to be --checkout?

Yes, that is how the `update = none` option is overriden

Reply:
> If we take out the clone you added to the ci.yml, we could test this?

Sure